extensions/kubernetes/vanilla/deployment/src/main/java/io/quarkus/kubernetes/deployment/KubernetesProcessor.java

```diff
@@ -74,6 +75,33 @@
 
     private static final String OUTPUT_ARTIFACT_FORMAT = "%s%s.jar";
 
+    @BuildStep
```

Comment:
how does these three methods interact....feels like it will be nondeterministic on who "wins" ?

Reply:
There is  not a race here.
We just produce build items,  based on the configuration properties that are found.